### ISSUE TITLE
Tajima's D values are positive around 4 for most windows

### ISSUE THREAD

**Issue by clairemerot** [ANGSD/angsd#236]
Hello, 
i'm getting suprinsingly high values of Tajima's D accross all the genome. I was expecting the baseline to be around 0 (was it right?) but it is rather around 4. Has anybody encoutered such kind of problem? is it worrisome or to be expected? My data has low coverage (1x per individual). Could it be that we overestimate heterozgotes?
Thanks for your insights!

![Thetas_ordered_TajDall](https://user-images.githubusercontent.com/28017150/60596750-a10a8100-9d77-11e9-92c4-49dca987827d.jpg)

**Comment by nspope** [comment]
What are the commands you are using to get these estimates? Are you applying any filters that would remove low-frequency alleles (e.g. -SNP_pval)? The expectation of Tajima's D under neutrality is 0, but it is a function of the SFS -- so any filter that distorts the SFS will bias Tajima's D. Removing low frequency alleles (e.g. singletons/doubletons) will substantially bias Tajima's D upward.

**Comment by clairemerot** [comment]
Thanks for your reply,
yes indeed we are removing low-frequency alleles because we are keeping only snps with maf >0.05. Do you think that would exert such a biais?
We also keep only position for which at least 50% of the individuals has a coverage of 1 but I don't think this should make a distrortion, shouldn't it?

Also, I have no ancestral genome so I use the reference for the -anc option but without the -fold option (this let me have values for thetas while with the fold option I had only nan - see in another issue). Could it explain the high values?

Cheers

**Comment by nspope** [comment]
Hi Claire,

TL;DR ... when estimating Tajima's D (or any SFS-based based stuff), don't use a minor frequency allele filter. A major reason to use ANGSD in the first place is so you can do this without worrying overmuch about biases caused by low sequencing depth/spurious genotype calls.

Removing low frequency alleles via a MAF filter is almost surely the cause of the upward bias in your estimates of D. It's easy to demonstrate  that under the neutral model (or really, *any* model or empirical SFS), removing low frequency bins results in a strong upward bias in D. 

In other words: estimates of D are absolute rubbish if you artificially alter the SFS by applying frequency-based filters. The same goes for *any* weighted function of the SFS, including Tajima's D and other statistics commonly used to detect departures from neutrality (Fu & Li's F, etc.). The same also goes for demographic reconstruction using methods like dadi (although there it is possible to modify the likelihood to account for the omitted bins.) Even when it is possible to modify a method or statistic to account for missing bins, this isn't desirable as a great deal of information is contained in the low frequency alleles regarding the demographic and selective processes that statistics like D are designed to be sensitive to. Instead, one of the major utilities of ANGSD is to estimate the SFS in its entirety while explicitly accounting for sequencing/genotyping error.

For a nice little introduction to the effect of frequency filters on D, see [this paper from Achaz](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2475743/). For a discussion of how D and related statistics are weighted functions of the SFS, see this [other paper by Achaz](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2746149/).

As for your second question: the lack of ancestral genome (& thus correct polarization of the SFS) doesn't matter in the slightest for this case. This is because the weights used by Tajima's D are symmetric -- you would get exactly the same estimate if you reversed the polarization (or if you used a folded SFS).

On the contrary, for estimating statistics that are not sensitive to low frequency alleles (like Fst), or for methods that characterize genetic structure without an explicit population genetic model (STRUCTURE-like stuff, PCA), it is often desirable to filter low frequency alleles because: (a) they may not provide any relevant information with regard to diferentiation, (b) they may introduce noise due to sequencing error; (c) they may violate statistical assumptions.

**Comment by clairemerot** [comment]
Thanks a lot for your detailled answer. Reading around, that now totally make sense! 
Sorry for my naive question :( 

Many thanks again, now everything is much much clearer regarding maf, folded spectrum, etc. I thought that I could mutualise the initial step that doSaf, do beagle, domaf in angsd (and then work on a subset of "sites" for all subsequent analyses) but I realise that this will need to be adjusted depending on the analysis (without filter, with filter, etc). I'll adjust our pipeline accordingly.

**Comment by z0on** [comment]
Hi Claire - I wrote this little review last year touching on ANGSD, SFS, and filtering concepts:  
https://matzlab.weebly.com/uploads/7/6/2/2/76229469/fantasticbeastssequence.pdf
It does not really add anything to Nate's great explanation, but perhaps it might be useful to further develop ANGSD'y intuition.

cheers
Misha

**Comment by clairemerot** [comment]
Great, thanks a lot Misha!
